packages/syft/src/syft/service/request/request.py

```diff
@@ -550,12 +592,19 @@ def valid(self) -> Union[SyftSuccess, SyftError]:
             )
         return SyftSuccess(message=f"{type(self)} valid")
 
-    def mutate(self, obj: UserCode, context: ChangeContext) -> Any:
-        res = obj.status.mutate(
-            value=self.value,
-            node_name=context.node.name,
-            verify_key=context.node.signing_key.verify_key,
-        )
+    def mutate(self, obj: UserCode, context: ChangeContext, revert: bool) -> Any:
+        if not revert:
+            res = obj.status.mutate(
+                value=self.value,
+                node_name=context.node.name,
+                verify_key=context.node.signing_key.verify_key,
+            )
+        else:
+            res = obj.status.mutate(
+                value=UserCodeStatus.DENIED,
```

Comment:
Currently, in case reverting a UserCodeStatusChange, I'm directly marking the status as denied. Since reverting of UserCodeStatusChange will only take place when we deny request, I think this should be fine for now. Although, we need to revisit this in 0.8.2. Let me know if you think otherwise ?